updates_and_signals/safe_message_handlers/activities.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+from dataclasses import dataclass
+from typing import List
+
+from temporalio import activity
+
+
+@dataclass
+class AllocateNodesToJobInput:
+    nodes: List[str]
+    job_name: str
+
+
+@activity.defn
+async def allocate_nodes_to_job(input: AllocateNodesToJobInput):
```

Comment:
Should provide type hints for every activity return, even if `-> None`, it helps callers